src/vm/eventpipeconfiguration.cpp

```diff
@@ -278,40 +277,83 @@ EventPipeProvider* EventPipeConfiguration::GetProviderNoLock(const SString &prov
     return NULL;
 }
 
+EventPipeSessionProvider* EventPipeConfiguration::GetSessionProvider(EventPipeSession *pSession, EventPipeProvider *pProvider)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        PRECONDITION(EventPipe::GetLock()->OwnedByCurrentThread());
+    }
+    CONTRACTL_END;
+
+    EventPipeSessionProvider *pRet = NULL;
+    if(pSession != NULL)
+    {
+       pRet = pSession->GetSessionProvider(pProvider);
+    }
+    return pRet;
+}
+
 size_t EventPipeConfiguration::GetCircularBufferSize() const
 {
     LIMITED_METHOD_CONTRACT;
 
-    return m_circularBufferSizeInBytes;
+    size_t ret = 0;
+    if(m_pSession != NULL)
+    {
+        ret = m_pSession->GetCircularBufferSize();
+    }
+    return ret;
 }
 
-void EventPipeConfiguration::SetCircularBufferSize(size_t circularBufferSize)
+EventPipeSession* EventPipeConfiguration::CreateSession(unsigned int circularBufferSizeInMB, EventPipeProviderConfiguration *pProviders, unsigned int numProviders)
```

Comment:
The EventPipeConfiguration doesn't add any of it's state to the session. Why do we route calls to create a session through EventPipeConfiguration?

Reply:
I think I had some initial reasoning for this, but it fell through as the code evolved. No reason at this time - it could be done in EventPipe directly.